test/Microsoft.AspNetCore.Server.KestrelTests/FrameTests.cs

```diff
@@ -23,10 +24,23 @@ public class FrameTests : IDisposable
     {
         private readonly SocketInput _socketInput;
         private readonly MemoryPool _pool;
-        private readonly Frame<object> _frame;
+        private readonly TestFrame<object> _frame;
         private readonly ServiceContext _serviceContext;
         private readonly ConnectionContext _connectionContext;
 
+        private class TestFrame<TContext> : Frame<TContext>
+        {
+            public TestFrame(IHttpApplication<TContext> application, ConnectionContext context)
+            : base(application, context)
+            {
+            }
+
+            public Task ProduceEndAsync()
+            {
+                return ProduceEnd();
```

Comment:
@halter73 Any strong feelings about doing this vs. just exposing `ProduceEnd` as public?